insecure/wintermute/attackOrchestrator.go

```diff
@@ -34,7 +56,244 @@ func (o *Orchestrator) WithAttackNetwork(network insecure.AttackNetwork) {
+			return fmt.Errorf("could not handle chunk data pack request event: %w", err)
+		}
+	case *messages.ChunkDataResponse:
+		if err := o.handleChunkDataPackResponseEvent(event); err != nil {
+			return fmt.Errorf("could not handle chunk data pack response event: %w", err)
+		}
+	default:
+		return fmt.Errorf("unexpected message type for wintermute attack orchestrator: %T", protocolEvent)
+	}
+
+	return nil
+}
+
+// corruptExecutionResult creates a corrupted version of the input receipt by tampering its content so that
+// the resulted corrupted version would not pass verification.
+func (o *Orchestrator) corruptExecutionResult(receipt *flow.ExecutionReceipt) *flow.ExecutionResult {
+	return &flow.ExecutionResult{
+		PreviousResultID: receipt.ExecutionResult.PreviousResultID,
+		BlockID:          receipt.ExecutionResult.BlockID,
+		// replace all chunks with new ones to simulate chunk corruption
+		Chunks:          unittest.ChunkListFixture(uint(len(receipt.ExecutionResult.Chunks)), receipt.ExecutionResult.BlockID),
+		ServiceEvents:   receipt.ExecutionResult.ServiceEvents,
+		ExecutionDataID: receipt.ExecutionResult.ExecutionDataID,
+	}
+}
+
+// handleExecutionReceiptEvent processes incoming execution receipt event from a corrupted execution node.
+// If no attack has already been conducted, it corrupts the result of receipt and sends it to all corrupted execution nodes.
+// Otherwise, it just bounces back the receipt to the sender.
+func (o *Orchestrator) handleExecutionReceiptEvent(receiptEvent *insecure.Event) error {
+	ok := o.corruptedIds.Contains(receiptEvent.CorruptedId)
+	if !ok {
+		return fmt.Errorf("")
+	}
+	if !ok {
+		return fmt.Errorf("could not find corrupted identity for: %x", receiptEvent.CorruptedId)
+	}
+
+	if corruptedIdentity.Role != flow.RoleExecution {
+		return fmt.Errorf("wrong sender role for execution receipt: %s", corruptedIdentity.Role.String())
+	}
+
+	receipt, ok := receiptEvent.FlowProtocolEvent.(*flow.ExecutionReceipt)
+	if !ok {
+		return fmt.Errorf("protocol event is not an execution receipt: %T", receiptEvent.FlowProtocolEvent)
+	}
+
+	lg := o.logger.With().
+		Hex("receipt_id", logging.ID(receipt.ID())).
+		Hex("executor_id", logging.ID(receipt.ExecutorID)).
+		Hex("result_id", logging.ID(receipt.ExecutionResult.ID())).
+		Hex("block_id", logging.ID(receipt.ExecutionResult.BlockID)).
+		Hex("corrupted_id", logging.ID(receiptEvent.CorruptedId)).
+		Str("protocol", insecure.ProtocolStr(receiptEvent.Protocol)).
+		Str("channel", string(receiptEvent.Channel)).
+		Uint32("targets_num", receiptEvent.TargetNum).
+		Str("target_ids", fmt.Sprintf("%v", receiptEvent.TargetIds)).Logger()
+
+	if o.state != nil {
+		// non-nil state means an execution result has already been corrupted.
+		if receipt.ExecutionResult.ID() == o.state.originalResult.ID() {
+			// receipt contains the original result that has been corrupted.
+			// corrupted result must have already been sent to this node, so
+			// just discards it.
```

Comment:
nit grammar: `just discard it.`

Reply:
https://github.com/onflow/flow-go/pull/2291/commits/f0555a9a48ce8f31c919b1e4ce76cea49a48f013